src/Admin/ContextAwareAdmin.php

```diff
@@ -43,22 +43,24 @@ public function getNewInstance()
     {
         $instance = parent::getNewInstance();
 
-        if ($contextId = $this->getPersistentParameter('context')) {
-            $context = $this->contextManager->find($contextId);
-
-            if (!$context) {
-                /** @var ContextInterface $context */
-                $context = $this->contextManager->create();
-                $context->setEnabled(true);
-                $context->setId($contextId);
-                $context->setName($contextId);
+        $contextId = $this->getPersistentParameter('context');
+        if (empty($contextId)) {
+            $contextId = ContextInterface::DEFAULT_CONTEXT;
+        }
 
-                $this->contextManager->save($context);
-            }
+        $context = $this->contextManager->find($contextId);
+        if (!$context) {
+            /** @var ContextInterface $context */
+            $context = $this->contextManager->create();
+            $context->setId($contextId);
+            $context->setName(ucfirst($contextId));
+            $context->setEnabled(true);
 
-            $instance->setContext($context);
+            $this->contextManager->save($context);
```

Comment:
Will this create a new context object in the database everytime you open create form? If yes could you move this to the `prePersist` section, so we only create a context object if we really have an entity.

Reply:
Yes, this is a good idea. I'll try.

Reply:
Ok @core23 this is too risky. Trying to comment $this->contextManager->save($context), lot of errors creating (for example, a tag). PrePersist isn't a choice because the context validation happens before.